kythe/cxx/indexer/cxx/GraphObserver.h

```diff
@@ -1070,6 +1071,10 @@ class GraphObserver {
   /// Note that the `VNameRef` should not outlive `id`.
   VNameRef DecodeMintedVName(const NodeId& id) const;
 
+  /// \brief Return a map of file IDs to metadata.
+  virtual const std::multimap<clang::FileID, std::shared_ptr<MetadataFile>>&
```

Comment:
minor: prefer `absl::flat_hash_map<clang::FileID, std::vector<...>>` unless this needs order (which doesn't appear to be the case)

Reply:
See below re: duplicate keys.

Reply:
So return a pair of iterators; exposing the entire map is unnecessary, especially since std::multimap is bad. Note: the possibility of duplicate keys is why I suggested `flat_hash_map<..., std::vector<...>>`

Reply:
Or even just a `std::vector<MetadataFile*>`; there's no need to provide callers with a strong reference or expose that to callers.

Reply:
Oh, I was so focused on the `flat_hash_map` I missed the `vector`.